test/unittests/CandidateTest.php

```diff
@@ -0,0 +1,750 @@
+     * Used to set behavior of NDB_Config test double
+     *
+     * @var array config name => value
+     */
+    private $_configMap = array();
+
+
+    /**
+     * Sets up fixtures:
+     *  - _candidate object
+     *  - config and Database test doubles
+     *  - _factory
+     *
+     * This method is called before each test is executed.
+     *
+     * @return void
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_configMap = array(
+                             array(
+                              'useProjects',
+                              false,
+                             ),
+                             array(
+                              'HeaderTable',
+                              null,
+                             ),
+                            );
+
+        $this->_listOfTimePoints = array(
+                                    array('ID' => '97'),
+                                    array('ID' => '98'),
+                                   );
+
+        $this->_configMock = $this->getMockBuilder('NDB_Config')->getMock();
+        $this->_dbMock     = $this->getMockBuilder('Database')->getMock();
+
+        $this->_factory   = NDB_Factory::singleton();
+        $this->_candidate = new Candidate();
+
+        $this->_factory->setConfig($this->_configMock);
+        $this->_factory->setDatabase($this->_dbMock);
+    }
+
+    /**
+     * Tears down the fixture, for example, close a network connection.
+     * This method is called after a test is executed.
+     *
+     * @return void
+     */
+    protected function tearDown()
+    {
+        parent::tearDown();
+        $this->_factory->reset();
+    }
+
+    /**
+     * Test select() method retrieves all _candidate and related info
+     *
+     * @return void
+     * @covers Candidate::select
+     * @covers Candidate::getData
+     * @covers Candidate::getListOfTimePoints
+     */
+    public function testSelectRetrievesCandidateInfo()
+    {
+        $this->_setUpTestDoublesForSelectCandidate();
+
+        $this->_candidate->select(969664);
+
+        //validate _candidate Info
+        $this->assertEquals($this->_candidateInfo, $this->_candidate->getData());
+
+        //validate list of time points
+        $expectedTimepoints = array();
+        foreach ($this->_listOfTimePoints as $oneRow) {
+            $expectedTimepoints[] = $oneRow['ID'];
+        }
+        $this->assertEquals(
+            $expectedTimepoints,
+            $this->_candidate->getListOfTimePoints()
+        );
+    }
+
+    /**
+     * Test select() method fails when invalid _candidate ID is passed
+     *
+     * @return void
+     * @covers Candidate::select
+     * @throws Exception
+     */
+    public function testsSelectFailsWhenInvalidCandidateIdPassed()
+    {
+
+        $this->_dbMock->expects($this->once())
+            ->method('pselectRow')
+            ->willReturn(false);
+
+        $this->setExpectedException('Exception');
```

Comment:
Maybe this should expect a more specific exception type than the base Exception class?